File: android/app/src/main/java/com/myfabricapp/UgoiraViewManager.java
package com.myfabricapp;

import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.facebook.react.bridge.ThemedReactContext;

public class UgoiraViewManager extends SimpleViewManager<UgoiraView> {
    @Override
    public String getName() {
        return "UgoiraView";
    }

    @Override
    protected UgoiraView createViewInstance(ThemedReactContext reactContext) {
        return new UgoiraView(reactContext);
    }

    @ReactProp(name = "paused")
    public void setPaused(UgoiraView view, boolean paused) {
        if (view != null) {
            if (paused) {
                view.pauseAnimation();
            } else {
                view.resumeAnimation();
            }
        }
    }
}

File: android/app/src/main/kotlin/com/myfabricapp/UgoiraViewManager.kt
package com.myfabricapp

import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.annotations.ReactProp
import com.facebook.react.bridge.ThemedReactContext

class UgoiraViewManager : SimpleViewManager<UgoiraView>() {
    override fun getName(): String {
        return "UgoiraView"
    }

    override fun createViewInstance(reactContext: ThemedReactContext): UgoiraView {
        return UgoiraView(reactContext)
    }

    @ReactProp(name = "paused")
    fun setPaused(view: UgoiraView, paused: Boolean) {
        if (paused) {
            view.pauseAnimation()
        } else {
            view.resumeAnimation()
        }
    }
}